### PR TITLE
Add doc describing how to contribute to project GIT repositories

### PR DESCRIPTION
**Pull request by berrange**
This page describes use of the issue tracker, commit sign off,
design docs and pull requests.

### CODE REVIEW

File: contrib-workflow.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+Depending on the repository, a number of automated tests will be performed on
+the submission, including a check that every commit has been signed off in
+accordance with the developer certificate of origin. Results of these automated
+tests will be displayed on the pull request summary page. If any failures are
+identified, the author is generally expected to resolve them and submit an
+updated pull request, unless the failure is caused by a flaw in the automated
+test system.
+
+Anyone with a github account is able to make comments and requests changes on
+pull requests, however, only members of the *contributors* team are able to
```
Comment (rmohr) [inline contrib-workflow.md]: Contributor in Github normally has the meaning of "having contributed some code". I think Maintainer is a better term.

File: contrib-workflow.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+Alternatively it is acceptable to submit the design doc as the first commit in
+a series of commits that implement the desired feature. The former will reduce
+the risk of code development taking an approach that is later rejected, while
+the later may allow faster development. Even once a design document is reviewed
+and merged, it is not mandatory to blindly follow it. If changes come to light
+during development that invalidate some of the design decisions, contributors
+should simply update the design doc to reflect the changed reality.
+
+
+## Developer certification
```
Comment (rmohr) [inline contrib-workflow.md]: Can't we just link to the DCO we have in kubevirt/kubevirt?

File: contrib.md [hunk 3]
```diff
@@ -25,6 +25,8 @@ with users in the forums, IRC channel, and any other appropriate venue.
 need for people to spread knowledge & passion about the project to potential
 new users & contributors.
 
+For contributions which involve changes to the project's GIT repositories,
```
Comment (rmohr) [inline contrib.md]: Can't we either link to https://github.com/kubevirt/kubevirt/blob/master/CONTRIBUTING.md and make sure that this applies to all kubevirt repos, or move that file here?

**Comment by berrange** [comment]
WRT the CONTRIBUTING.md file in the kubevirt repo, I'd expect to update that content to a large degree. In particular take out the generic workflow stuff, and just point to this webpage instead. In its place put more technical details that are specific to the kubevirt repo codebase.

File: contrib-workflow.md (hunk 2 above)
Comment (berrange) [inline contrib-workflow.md]: I think it is preferable to have it included in both places. In particular this is something that applies to the project as a whole, not just one git repo.

File: contrib-workflow.md (hunk 1 above)
Comment (berrange) [inline contrib-workflow.md]: I was literally referring to the team name shown in the github organization, but I made a mistake. I did indeed mean the 'maintainer' team. I should also hyperlink it to the team member list page in github

**Comment by berrange** [comment]
Refreshed to clarify some comments, and add further hyperlinks.

File: contrib-workflow.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+design for the implementation. Instead it just serves as a place to consider
+the acceptability of the feature request at a high level.
+
+If a person is actively working on an issue, it is expected that they set
+themselves as the assignee for it, to prevent others accidentally duplicating
+effort. Conversely if an assignee has to stop their work for whatever reason,
+it is polite to either unassign themselves from the issue, or put a comment
+on it mentioning the pause in activity. This gives a chance for other interested
+contributors to pick up the work. If a contributor does not have permission to
+set themselves as assingee, they can add a comment to the issue asking to be
```
Comment (fabiand) [inline contrib-workflow.md]: I'd pull this last sentence further up, as I expect that newcomers (who can not assign themselves) will be the ones reading this guide.

File: contrib-workflow.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+made the assignee.
+
+When fixing an issue, the commit message of the patch that provides the fix
+must mention the issue tracker number. The pull request may also mention the
```
Comment (fabiand) [inline contrib-workflow.md]: Must mention the issue number, as well as one of th keywords to auto-close them, see https://help.github.com/articles/closing-issues-via-commit-messages/

File: contrib-workflow.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+on software deliverables, such as the primary
+"[kubevirt](https://github.com/kubevirt/kubevirt)" repository and the
+"[project-infra](https://github.com/kubevirt/project-infra)" repository.
+Repositories serving other purposes such as the
+"[kubvirt.github.io](https://github.com/kubevirt/kubevirt.github.io)" website
+content or the "community" content, do not make use of design documentation.
+
+There is no mandatory requirement to get a design document approved and merged
+before starting on the code development. Contributors can choose to submit a
+design doc upfront and get it reviewed and merged before starting coding.
```
Comment (fabiand) [inline contrib-workflow.md]: Worth noting that it's recommended to submit design documents up front for controversial or complex topics.

File: contrib.md (hunk 3 above)
Comment (fabiand) [inline contrib.md]: I do see the CONTRIBUTING as a repos specific addition to the contrib-worklflow document.

To me we can see over the mid-run how to align them to each other.

**Comment by berrange** [comment]
Refreshed with updates for comments.

**Comment by rmohr** [comment]
As stated before, I think this document is way too detailed. It completely hides the fact that you just do the normal github thing + the proposal. I don't want to be destructive and I really think it is very accurate, but I think it makes it harder instead of easier for people to contribute.

I think just saying
 1) Hey we love contributions!
 2) Use your usual github workflow. When unsure, just /share, ask or file an issue, we will sort things out.
 3) DCO instead of CLA
 4) make a doc if you want to propose something

is much better.

**Comment by berrange** [comment]
The concise instructions you describe are shown right at the start of the doc " ## Process summary " - just a short bullet point list there.  The rest of the doc is about giving the full details for those who want to understand more, which *is* important IMHO, because not everyone coming to the project is going to have the knowledge about "normal github" work.

**Comment by fabiand** [comment]
Sounds like this PR contains what @rmohr requested, and more.
Also the content is correct and of value. Thus merging it.

**Pull request merged**